analyzer/evmverifier/sourcify/client.go

```diff
@@ -0,0 +1,170 @@
+//
+// Note: This uses the free, public server API. If it turns out to be unreliable, we could use the repository API (vis IPFS proxy) instead, e.g.:
+// http://ipfs.default:8080/ipns/repo.sourcify.dev/contracts/full_match/23294
+func (s *SourcifyClient) GetVerifiedContractAddresses(ctx context.Context, runtime common.Runtime) ([]ethCommon.Address, error) {
+	// Fetch verified contract addresses.
+	u := *s.serverUrl
+	u.Path = path.Join(u.Path, "files/contracts", sourcifyChains[s.chain][runtime])
+	body, err := s.callAPI(ctx, http.MethodGet, &u)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch verified contract addresses: %w (%s)", err, u.String())
+	}
+
+	// Parse response.
+	var response struct {
+		Full []ethCommon.Address `json:"full"`
+		// XXX: Skip partial matches for now: https://docs.sourcify.dev/docs/full-vs-partial-match/
+	}
+	if err := json.Unmarshal(body, &response); err != nil {
+		return nil, fmt.Errorf("failed to parse verified contract addresses: %w (%s)", err, u.String())
+	}
+
+	return response.Full, nil
+}
+
+// GetContractSourceFiles returns the source files for the given contract address that is verified on sourcify.
+// The first return argument are the source files returned in a list of maps, where each map contains the following keys:
+// - "name": The name of the source file.
+// - "content": The content of the source file.
+// - "path": The path of the source file.
+// For convenience the second return argument is the metadata.json file content as a raw JSON message.
+//
+// See https://docs.sourcify.dev/docs/api/server/get-source-files-full/ for more details.
+//
+// Note: This uses the free, public server API. If it turns out to be unreliable, we could use the repository API (vis IPFS proxy) instead, e.g.:
+// - https://docs.sourcify.dev/docs/api/repository/get-file-repository/
+// - http://ipfs.default:8080/ipns/repo.sourcify.dev/contracts/full_match/23294/0x0a0b58b5e6d8f2c0f4c4b6e7a0c8f0b1b4b3b2b1/
+func (s *SourcifyClient) GetContractSourceFiles(ctx context.Context, runtime common.Runtime, address ethCommon.Address) ([]map[string]json.RawMessage, json.RawMessage, error) {
+	// Fetch contract source files.
+	u := *s.serverUrl
+	u.Path = path.Join(u.Path, "files", sourcifyChains[s.chain][runtime], address.String())
+	body, err := s.callAPI(ctx, http.MethodGet, &u)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch contract source files: %w (%s)", err, u.String())
+	}
+
+	// Ensure response is a valid JSON.
+	var response []map[string]json.RawMessage // Response is an array of source files.
+	if err = json.Unmarshal(body, &response); err != nil {
+		return nil, nil, fmt.Errorf("failed to parse contract source files: %w (%s)", err, u.String())
+	}
+
+	// Find metadata.json.
+	var metadata json.RawMessage
+	var found bool
+	for _, file := range response {
+		var name string
+		if err = json.Unmarshal(file["name"], &name); err != nil {
+			continue
+		}
+		if name == "metadata.json" {
```

Comment:
Let's not include metadata.json in the `response` variable. WDYT? Two reasons:
- `response` should be about source files, and metadata.json is really more of a byproduct of compilation, not a source (even though it does reflect some source compilation inputs, like the compiler flags). 
- It's wasteful and potentially confusing to duplicate metadata.json in storage, API responses, etc. (Once as a special "metadata" value and once as a source file.)

It looks to me that its being included with the source files is just due to Sourcify (ab)using the source-files endpoint to keep things simple, not due to some deeper reason.

Reply:
Yeah seems fine with me. Removed it.